src/main/java/com/undertheriver/sgsg/foler/service/FolderService.java

```diff
@@ -88,4 +98,44 @@ public void delete(Long folderId) {
             .orElseThrow(ModelNotFoundException::new);
         folder.delete();
     }
+
+    public void secret(Long userId, Long folderId) {
+        Folder folder = folderRepository.findById(folderId)
+            .orElseThrow(ModelNotFoundException::new);
+
+        validateUserHasFolder(userId, folder);
+
+        folder.secret();
+    }
+
+    public void unsecret(Long userId, Long folderId, FolderDto.UnsecretReq request) {
+        validateFolderPassword(userId, request.getPassword());
+
+        Folder folder = folderRepository.findById(folderId)
+            .orElseThrow(ModelNotFoundException::new);
+
+        validateUserHasFolder(userId, folder);
+
+        folder.unsecret();
+    }
+
+
+    private void validateFolderPassword(Long userId, String rawPassword) {
+        User user = userRepository.findById(userId)
+            .orElseThrow(ModelNotFoundException::new);
+
+        if (!user.hasFolderPassword()) {
+            throw new PasswordValidationException(NO_PASSWORD);
+        }
+
+        if (!bCryptPasswordEncoder.matches(rawPassword, user.getFolderPassword())) {
+            throw new PasswordValidationException(PASSWORD_NOT_MATCH);
+        }
+    }
+
+    private void validateUserHasFolder(Long userId, Folder folder) {
+        if (!folder.ownBy(userId)) {
+            throw new BadRequestException(UNMACHED_USER);
+        }
```

Comment:
```suggestion
        if (!folder.hasBy(userId)) {
            throw new BadRequestException(UNMACHED_USER);
        }
```
자바에서 bool을 리턴하는 메서드는 is, has 등의 네이밍을 사용합니다
https://stackoverflow.com/questions/3874350/naming-conventions-for-java-methods-that-return-boolean